dodiscover/ci/clf_test.py

```diff
@@ -0,0 +1,359 @@
+from typing import Callable, Optional, Set, Tuple, Union
+
+import numpy as np
+import pandas as pd
+import scipy.special
+import sklearn
+import sklearn.metrics
+from numpy.typing import NDArray
+from sklearn.neighbors import NearestNeighbors
+from sklearn.utils import check_random_state
+
+from dodiscover.typing import Column
+
+from .base import BaseConditionalIndependenceTest
+
+
+class ClassifierCITest(BaseConditionalIndependenceTest):
+    def __init__(
+        self,
+        clf: sklearn.base.BaseEstimator,
```

Comment:
Can we make this compatible with a deep learning based classifier?

Reply:
I changed the type and also added it in the docstring. I think adding an example would definitely be helpful. I'm not as familiar with the torch/keras/tensorlfow API as I am with scikit-learn. This could be a good example for a new contributor.

I started an issue.